web/modules/custom/dbcdk_community/src/Plugin/Block/ProfilesBlock.php

```diff
@@ -301,6 +283,52 @@ protected function getQuarantinedProfiles(array $profiles_filter) {
   }
 
   /**
+   * Count all quarantined profiles matching a filter.
+   *
+   * @param array $profiles_filter
+   *   An array containing filter arguments.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   *
+   * @return int|NULL
+   *   The number of quarantined profiles matching the filter.
+   */
+  protected function getQuarantinedProfilesCount(array $profiles_filter = []) {
+    // Fetch all quarantines that are active from the moment of the request.
+    $quarantined_filter = [
+      'where' => [
+        'end' => [
+          // The string should be in a format recognized by the Date.parse()
+          // method which is ISO-8601 in our case.
+          // @See https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Date
+          'gt' => date(\DateTime::ATOM),
+        ],
+      ],
+    ];
+    $quarantines = (array) $this->quarantineApi->quarantineFind(json_encode($quarantined_filter));
+
+    // Reduce all the quarantines to an array of quarantined profile ids.
+    // We do this to make sure a profile id only appears once since we use these
+    // values as "Where" arguments for the next request to the service, and a
+    // profile can have multiple quarantines at the same time.
+    $quarantine_ids = array_reduce($quarantines, function($result, Quarantine $quarantine) {
+      $result[$quarantine->getQuarantinedProfileId()] = ['id' => $quarantine->getQuarantinedProfileId()];
+      return $result;
+    });
+
+    // Reset array keys so json_encode() will handle the ids as an indexed array
+    // and not an associative array.
+    $quarantine_ids = array_values($quarantine_ids);
+
+    // Use the array of quarantined ids as "where" arguments.
+    $profiles_filter['or'] = $quarantine_ids;
+
+    $result = $this->profileApi->profileCount(json_encode($profiles_filter));
+    return (!empty($result['count'])) ? $result['count'] : NULL;
```

Comment:
Why do we return NULL instead of 0? It would keep the return type an integer at all times instead of a mixture